packages/react-dom/src/__tests__/ReactServerRenderingHydration-test.js

```diff
@@ -660,14 +660,45 @@ describe('ReactDOMServerHydration', () => {
     document.body.removeChild(parentContainer);
   });
 
-  it('regression test: Suspense + hydration in legacy mode ', () => {
+  it('Suspense + hydration in legacy mode', () => {
     const element = document.createElement('div');
     element.innerHTML = '<div>Hello World</div>';
+    let div = element.firstChild;
+    let ref = React.createRef();
+    expect(() =>
+      ReactDOM.hydrate(
+        <React.Suspense fallback={null}>
+          <div ref={ref}>Hello World</div>
+        </React.Suspense>,
+        element,
+      ),
+    ).toWarnDev(
+      'Warning: Did not expect server HTML to contain a <div> in <div>.',
```

Comment:
This warns because of the superfluous div but before that it should warn about missing a Suspense boundary but it's currently missing that warning:

https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactDOMHostConfig.js#L843